instrumentation/sidekiq/lib/opentelemetry/instrumentation/sidekiq/instrumentation.rb

```diff
@@ -67,24 +67,24 @@ def patch_on_startup
         def add_client_middleware
           ::Sidekiq.configure_client do |config|
             config.client_middleware do |chain|
-              chain.add Middlewares::Client::TracerMiddleware
+              chain.prepend Middlewares::Client::TracerMiddleware
```

Comment:
How do we prevent this from regressing? Is there a automated test you would be able to submit with this to ensure that the module is the first in the ancestry?

How do we ensure it is always the first module in the list?

Reply:
> How do we prevent this from regressing? Is there a automated test you would be able to submit with this to ensure that the module is the first in the ancestry?

Because the mechanism of actually invoking the `install` block, which is what causes us to do the "install", depends on the machinery of a single `OpenTelemetry::SDK`, I'm not sure we can write a fully integrated test. But if we're OK trusting `OpenTelemetry::Instrumentation::Base::install` and `#install` to be stable and a public API, then yeah, I think I can work something out. Though I expect it's going to require some nasty `instance_variable_set`'s to reset the Sidekiq configuration state between tests.

> How do we ensure it is always the first module in the list?

In the general case, I'm not sure we can as we don't have a way to know when everyone is done configuring the Sidekiq middleware chains. In Rails-land, _maybe_ we could use a Railtie to wire up an after initialize hook to check. But even then, I'm not sure we'd want to as it's conceivable that someone has a use case where they need something else going even higher in the chain. Generic exception handling middleware (Bugsnag, Rollbar, etc… come to mind).

Reply:
OK, I've added some tests - including some hackery to work with Sidekiq 5, 6, and 7. Phew!